cdc/processor.go

```diff
@@ -937,6 +948,29 @@ func (p *processor) addTable(ctx context.Context, tableID int64, replicaInfo *mo
 						lastResolvedTs = pEvent.CRTs
 						p.localResolvedNotifier.Notify()
 						resolvedTsGauge.Set(float64(oracle.ExtractPhysical(pEvent.CRTs)))
+
+						// We need to wait for the globalResolvedTs to be updated,
+						// because we may be a newly recovered table and the localResolvedTs
+						// has just been reset to the previous checkpoint.
+						// Waiting will ensure that any premature flush sink requests have been sent
+						// and that whatever we will send will not be interrupted by sink flush requests
+						// driven by an outdated globalResolvedTs.
+						for {
+							gTs := atomic.LoadUint64(&p.globalResolvedTs)
+							if gTs <= lastResolvedTs {
```

Comment:
I don't quite understand the logic, if `gTs > lastResolvedTs` (which means the table is newly added and resovled reset to previous checkpoint), should we wait the `lastResolvedTs` reaches the `gTs`? but it seems the for loop will prevent the update of `lastResolvedTs`